src/Compilers/CSharp/Portable/Parser/Lexer.cs

```diff
@@ -2220,6 +2220,10 @@ private void LexSyntaxTrivia(bool afterFirstToken, bool isTrailing, ref SyntaxLi
                         this.AddTrivia(this.ScanWhitespace(), ref triviaList);
                         break;
                     case '/':
+                        // PERF: This pools all the comments that are at the top of the file, i.e. file copyright headers.
+                        // Most of them repeat themselves and it can lead to a noticeable amount of string duplicates for huge projects.
+                        int maximumInternLength = afterFirstToken ? -1 : 256;
```

Comment:
can we actually look at some common copyright clauses.  For example, the gpl is:

```
    This program is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.

    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License
    along with this program.  If not, see <http://www.gnu.org/licenses/>.
```

This is about 700 chars.

If you had this in every file (in a /**/) block, then it wouldn't be included.  Maybe that's ok?  Maybe we just have ths optimization for //-style comments?  

IMO, i think nearly all leading comments per file will be copyright.   So, as long as we have some cap (so nothing pathological sneaks through), i think it's reasonable to grab things larger than 256 chars.  Maybe 2k so that apache/gnu/etc. are all handled?

Reply:
That would be a good idea too.